### PR TITLE
Add pagination return and ability to pass query to SP list's get_items().

### PR DESCRIPTION
**Pull request by hcallen**
I found these features to be necessary while working with a rather large list.

### CODE REVIEW

**Comment by hcallen** [comment]
> Can you explain the change in [81af390](https://github.com/O365/python-o365/commit/81af3908fe31bb8f17adea3103340c73401cfd24)

It allows the ability include both expand and select when querying SharePoint list items. I found it necessary when attempting to get a Lookup field (Person) instead of a LookupId field.

See examples at:
[Enumerate items in a list](https://docs.microsoft.com/en-us/graph/api/listitem-list?view=graph-rest-1.0&tabs=http)
[FieldValueSet resource](https://docs.microsoft.com/en-us/graph/api/resources/fieldvalueset?view=graph-rest-1.0)

File: O365/sharepoint.py [hunk 1]
```diff
@@ -505,6 +505,22 @@ def get_list_by_name(self, display_name):
...
+        Creates a SharePoint list.
+        :param list_data: Dict representation of list.
+        :type list_data: Dict
+        :rtype: list[SharepointList]
+        """
+        url = self.build_url(self._endpoints.get('get_lists'))
+        response = self.con.post(url, data=list_data)
+
+        if not response:
+            return []
```
[the remark this answers is not in the document]
Comment (hcallen) [reply O365/sharepoint.py]: Performed update.

File: O365/sharepoint.py [hunk 2]
```diff
@@ -168,6 +169,13 @@ def save_updates(self):
         self._clear_tracker()
         return True
 
+    def delete(self):
+        url = self.build_url(self._endpoints.get('delete_list_item').format(item_id=self.object_id))
+        response = self.con.delete(url)
+        if not response:
```
[the remark this answers is not in the document]
Comment (hcallen) [reply O365/sharepoint.py]: Performed update.

**Comment by janscas** [comment]
Thanks

**Pull request merged**